javascript/modules/client/src/client-erc20.ts

```diff
@@ -1,703 +1,52 @@
 import {
-  Erc20Proposal,
-  Erc20ProposalListItem,
-  Erc20TokenDetails,
-  ExecuteProposalStep,
-  ExecuteProposalStepValue,
-  ICanVoteParams,
   IClientErc20,
-  ICreateProposalParams,
+  IClientErc20Decoding,
+  IClientErc20Encoding,
+  IClientErc20Estimation,
+  IClientErc20Methods,
   IErc20PluginInstall,
-  IExecuteProposalParams,
-  IMintTokenParams,
-  IPluginSettings,
-  IProposalQueryParams,
-  IVoteProposalParams,
-  ProposalCreationSteps,
-  ProposalCreationStepValue,
-  ProposalMetadata,
-  ProposalSortBy,
-  SubgraphErc20Proposal,
-  SubgraphErc20ProposalListItem,
-  VoteProposalStep,
-  VoteProposalStepValue,
-  VoteValues,
 } from "./internal/interfaces/plugins";
-import { IDAO } from "@aragon/core-contracts-ethers";
-import { ClientCore } from "./internal/core";
-import {
-  DaoAction,
-  GasFeeEstimation,
-  IInterfaceParams,
-  IPluginInstallItem,
-  SortDirection,
-} from "./internal/interfaces/common";
+import { IClientErc20MethodsModule } from "./modules-erc20/methods";
 import { ContextPlugin } from "./context-plugin";
-import {
-  computeProposalStatusFilter,
-  isProposalId,
-  toErc20Proposal,
-  toErc20ProposalListItem,
-} from "./internal/utils/plugins";
-import {
-  decodeMintTokenAction,
-  decodeUpdatePluginSettingsAction,
-  encodeErc20ActionInit,
-  encodeMintTokenAction,
-  encodeUpdatePluginSettingsAction,
-  getFunctionFragment,
-} from "./internal/encoding/plugins";
-import {
-  bytesToHex,
-  GraphQLError,
-  InvalidAddressError,
-  InvalidAddressOrEnsError,
-  InvalidProposalIdError,
-  NoProviderError,
-  Random,
-} from "@aragon/sdk-common";
-import { delay } from "./internal/temp-mock";
-import { isAddress } from "@ethersproject/address";
-import {
-  QueryErc20Proposal,
-  QueryErc20Proposals,
-} from "./internal/graphql-queries/proposal";
-import { formatEther } from "@ethersproject/units";
-import { QueryErc20PluginSettings } from "./internal/graphql-queries/settings";
-import { QueryToken } from "./internal/graphql-queries/token";
+import { IClientErc20EncodingModule } from "./modules-erc20/encoding";
+import { IClientErc20DecodingModule } from "./modules-erc20/decoding";
+import { IClientErc20EstimationModule } from "./modules-erc20/estimation";
```

Comment:
Do we need `Module` in the name? same as above

Reply:
See above